src/main/java/com/bobocode/svydovets/context/AnnotationConfigurationApplicationContext.java

```diff
@@ -1,77 +1,136 @@
+import java.util.Objects;
 import java.util.stream.Collectors;
 
+import org.apache.commons.lang3.ClassUtils;
 import org.apache.commons.lang3.StringUtils;
 
+import com.bobocode.svydovets.annotation.Inject;
 import com.bobocode.svydovets.beans.factory.BeanFactory;
 import com.bobocode.svydovets.beans.factory.DefaultListableBeanFactory;
 import com.bobocode.svydovets.beans.scanner.ComponentBeanScanner;
 import com.bobocode.svydovets.exception.NoSuchBeanDefinitionException;
 import com.bobocode.svydovets.exception.NoUniqueBeanDefinitionException;
 
 public class AnnotationConfigurationApplicationContext implements ApplicationContext {
-    private Map<String, Object> nameToBean;
+    private Map<String, Object> beanContainer;
     private final BeanFactory beanFactory;
     private final ComponentBeanScanner componentScanner;
 
     public AnnotationConfigurationApplicationContext(String packageName) {
         this.beanFactory = new DefaultListableBeanFactory();
         this.componentScanner = new ComponentBeanScanner();
-        this.scan(packageName);
-    }
-
-    public AnnotationConfigurationApplicationContext(BeanFactory beanFactory,
-                                                     ComponentBeanScanner componentScanner,
-                                                     String packageName) {
-        this.beanFactory = beanFactory;
-        this.componentScanner = componentScanner;
-        this.scan(packageName);
+        init(packageName);
     }
 
     @Override
     public <T> T getBean(Class<T> beanType) throws NoSuchBeanDefinitionException, NoUniqueBeanDefinitionException {
-        var beans = nameToBean.values().stream()
-                .filter(beanType::isInstance)
-                .toList();
+        Objects.requireNonNull(beanType, "The beanType cannot be null!");
+
+        var beans = beanContainer.values().stream()
+          .filter(beanType::isInstance)
+          .toList();
+
         if (beans.size() > 1) {
-            throw new NoUniqueBeanDefinitionException();
+            throw new NoUniqueBeanDefinitionException(String.format("The Bring contains %s beans of the %s. Please, "
+              + "specify a beanName", beans.size(), beanType.getSimpleName()));
         }
         if (beans.isEmpty()) {
-            throw new NoSuchBeanDefinitionException();
+            throw new NoSuchBeanDefinitionException(String.format("Bean with type %s is not found",
+              beanType.getSimpleName()));
         }
 
         return beanType.cast(beans.get(0));
     }
 
     @Override
-    public Object getBean(String name) throws NoSuchBeanDefinitionException {
-        return nameToBean.get(name);
+    public Object getBean(String beanName) throws NoSuchBeanDefinitionException {
+        Objects.requireNonNull(beanName, "The beanName cannot be null!");
+
+        if (StringUtils.isEmpty(beanName)) {
+            throw new IllegalArgumentException("Bean name is empty! Please specify the bean name.");
```

Comment:
I think it is better to `beanName` as the parameter name. As you use it for not null checks.